SpaceDuckFrontend/szachy/src/views/Szachy.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import Header from '../components/Header'
+import './szachy-styles.css';
+import cosmoDuck from '../assets/Cosmo_duck.png'
+import address from '../configuration.json';
+import Cookies from 'universal-cookie';
+const cookies = new Cookies();
+class Szachy extends React.Component {
+    constructor() {
+
+        super();
+        this.state = {
+            guest: false,
+            instructionPopup:false,
+            guestName:''
+        }
+       this.playAsGuest = this.playAsGuest.bind(this);
+       this.handleInstruction = this.handleInstruction.bind(this);
+       this.handleCloseInstruction = this.handleCloseInstruction.bind(this);
+       this.handleChange = this.handleChange.bind(this);
+       this.goToMainService = this.goToMainService.bind(this);
+       this.saveGuestName  = this.saveGuestName.bind(this);      
+    }
+
+    playAsGuest(){
+        this.setState({guest:true});
+    }
+    handleInstruction(){
+    	this.setState({instructionPopup:true});
+    }
+    handleCloseInstruction(){
+    	this.setState({instructionPopup:false});
+    }
+    handleChange(event) {
+        this.setState({guestName: event.target.value});    
+    }
+     goToMainService(){
+       cookies.set('game', 'szachy', { path: '/' });
+    }
+    unLogged(){
+    	return(
+    		<div className="asGuest">
+    			<a href={"http://"+address.baseURL+":"+address.mainPort+"/login"} onClick={this.goToMainService} className="button inline-button"> Zaloguj </a>
+             	<a href="#" className="button inline-button" onClick={this.playAsGuest} > Gość </a><br/>
+             			{this.state.guest ? <input type="text" placeholder="imię" onChange={this.handleChange}/>:null}
+    		</div>
+    		)
+    }
+    saveGuestName(){
+    	if(this.state.guest==true){
+    		localStorage.setItem('guest', this.state.guestName);
+    	};
+    }
+    showInstructionsPopup(){
+    	return(
+    		<div className="overlay">
+    			<div className="instructionPopup">	
+    				<a className="close" href='#' onClick={this.handleCloseInstruction}>X</a>
+    				<p><b>Szachy</b><br/><br/>W grze udział biorą dwie osoby. Rozgrywa się ją na planszy nazywanej szachownicą, gdzie rozstawia się 32 bierki (sześciu rodzajów, w tym po 8 pionów i 8 figur, łącznie 16 dla każdego z graczy). Celem gry jest danie mata, tzn. zagrożenie królowi przeciwnika usunięciem z dalszej rozgrywki („zbiciem”), którego nie sposób uniknąć. 
```

Comment:
Takie kawałki kodu najlepiej by było trzymać w jakichś osobnych plikach, np. w jakimś jsonie. Na to już pewnie nie ma za bardzo czasu teraz, szczególnie że inne serwisy już działają, ale może warto mieć to na uwadze na przyszłość.